rdm/extensions.py

```diff
@@ -0,0 +1,66 @@
+from importlib import import_module
+
+from jinja2 import nodes
+from jinja2.ext import Extension
+
+
+class RdmExtension(Extension):
+
+    def __init__(self, environment):
+        super().__init__(environment)
+
+        # add the post processing list to the environment.
+        if not hasattr(environment, 'rdm_post_process_filters'):
+            environment.extend(rdm_post_process_filters=[])
+
+    @staticmethod
+    def post_processing_filter_list(environment):
+        return getattr(environment, 'rdm_post_process_filters', [])
+
+    # Check if there is a post_process_filter method.
+    # If so add it to the post filtering list if it is not already added.
+    def check_add_post_filter(self):
+        post_filter = getattr(self, 'post_process_filter', None)
+        current_post_filters = self.post_processing_filter_list(self.environment)
+        if post_filter and post_filter not in current_post_filters:
+            current_post_filters.append(post_filter)
+
+    # default parser sets up a call back that will add a post process filter if one exists.
```

Comment:
Could you explain how this `parse` method and `block_callback` works, and why they are necessary?

Reply:
Parse happens when the syntax tree of nodes is being built. The call back happens when the tree is being walked during generation. It is skipped inside of a false if. That branch is not walked.